code/Extension/UserFormFileExtension.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace SilverStripe\UserForms\Extension;
+
+use SilverStripe\Assets\File;
+use SilverStripe\Assets\Folder;
+use SilverStripe\ORM\DataExtension;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\Queries\SQLUpdate;
+use SilverStripe\UserForms\Control\UserDefinedFormController;
+use SilverStripe\UserForms\Model\Submission\SubmittedFileField;
+use SilverStripe\Versioned\Versioned;
+
+class UserFormFileExtension extends DataExtension
+{
+
+    public const USER_FORM_UPLOAD_UNKNOWN = null;
+
+    public const USER_FORM_UPLOAD_FALSE = 'f';
+
+    public const USER_FORM_UPLOAD_TRUE = 't';
+
+    private static $db = [
+        'UserFormUpload' => "Enum('f, t', null)",
+    ];
+
+    /**
+     * Check if the file is associated with a userform submission
+     * Save the result in the database as a tri-state for two reasons:
+     * a) performance - prevent the need for an extra DB query
+     * b) if in the future the userform submission is deleted and the uploaded file is not (file is orphaned),
+     *    then it is still recorded that the file was originally uploaded from a userform submission
+     *
+     * @param bool $value
+     * @see File::isTrackedFormUpload(), UserDefinedFormController::process()
+     */
+    public function updateTrackedFormUpload(&$value): void
+    {
+        /** @var File|Versioned $file */
+        $file = $this->owner;
+        if ($file->UserFormUpload != self::USER_FORM_UPLOAD_UNKNOWN) {
+            $value = $file->UserFormUpload == self::USER_FORM_UPLOAD_TRUE;
+            return;
+        }
+        if ($file->ClassName == Folder::class) {
+            $value = false;
+        } else {
+            $value = SubmittedFileField::get()->find('UploadedFileID', $file->ID) ? true : false;
+        }
+        $this->updateDB($value);
+    }
+
+    /**
+     * Update File.UserFormUpload draft table without altering File.LastEdited
+     *
+     * @param bool $value
+     */
+    private function updateDB(bool $value): void
+    {
+        if (!$this->owner->isInDB()) {
+            return;
+        }
+        $table = DataObject::getSchema()->tableName(File::class);
+        $column = 'UserFormUpload';
+        $enumVal = $value ? self::USER_FORM_UPLOAD_TRUE : self::USER_FORM_UPLOAD_FALSE;
+        SQLUpdate::create()
```

Comment:
Why are we circumventing the model layer here and tightly coupling to the database schema in this way?

Reply:
Want to update an new and unpopulated value in the database without touching the LastEdited time, add a version, etc

I'm not aware of a way to go about doing a 'stealth update' on the database via the ORM

The only other ways that I know of populating new db columns would be queuedjobs or requireDefaultRecords, though I want to avoid both of those

Reply:
I'm also unaware of any other methods for updating records without triggering changes to the LastEdited column - I'm surprised it's not an available flag on `DataObject::write()`.

The hardcoded column name isn't ideal, _perhaps_ this could be a class constant that's used in the `$db` array? Otherwise I think this can move forward as-is for now.

Reply:
It's fairly minor and very low-risk, will just keep as is